src/tests/ftest/osa/osa_dmg_negative_test.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+from osa_utils import OSAUtils
+from test_utils_pool import TestPool
+
+
+class OSADmgNegativeTest(OSAUtils):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description: This test
+    passes invalid parameters to dmg OSA
+    related commands and check whether
+    appropriate errors are returned.
+
+    :avocado: recursive
+    """
+    def setUp(self):
+        """Set up for test case."""
+        super(OSADmgNegativeTest, self).setUp()
+        self.dmg_command = self.get_dmg_command()
+        # Start an additional server.
+        self.extra_servers = self.params.get("test_servers",
+                                             "/run/extra_servers/*")
+        # Dmg command test seeuence
+        self.test_seq = self.params.get("dmg_cmd_test",
+                                        "/run/dmg_cmds/*")
+
+    def validate_results(self, exp_result, dmg_output):
+        """Validate the dmg_output results with expected results
+
+        Args:
+            exp_result (str) : Expected result (Pass or Fail string).
+            dmg_output (str) : dmg output string.
+        """
+        if exp_result == "Pass":
+            self.is_rebuild_done(3)
+            if "succeeded" in dmg_output:
+                self.log.info("Test Passed")
+            else:
+                self.fail("Test Failed")
+        elif exp_result == "Fail":
+            if "failed" in dmg_output:
+                self.log.info("Test Passed")
+            else:
+                self.fail("Test Failed")
+        else:
+            self.fail("Invalid Parameter")
+
+    def run_osa_dmg_test(self, num_pool):
+        """Run the offline extend without data.
+
+        Args:
+            num_pool (int) : total pools to create for testing purposes.
+        """
+        # Create a pool
+        self.dmg_command.exit_status_exception = False
+        pool = {}
+        pool_uuid = []
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context, dmg_command=self.dmg_command)
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+            self.pool = pool[val]
+
+        # Start the additional servers and extend the pool
+        self.log.info("Extra Servers = %s", self.extra_servers)
+        self.start_additional_servers(self.extra_servers)
+        # Give sometime for the additional server to come up.
+        time.sleep(5)
```

Comment:
Can you call dmg system query every few sec to see if the additional server has come up instead of waiting for the fixed 5 sec?

Reply:
Sleep not required... start_additional_servers should take care of handling that the server is up.